### ISSUE TITLE
ActiveSupport::Duration#iso8601 number with e notation

### ISSUE THREAD

**Issue by stanig2106** [rails/rails#45113]
### Steps to reproduce

```ruby
big_number = 1_000_000_000

iso_duration_string = big_number.seconds.iso8601 # "PT1e+09S"

parsed_duration = ActiveSupport::Duration.parse(iso_duration_string) 
```

### Expected behavior

The `iso_duration_string` should be equal to `"PT1000000000S"`
or 
`ActiveSupport::Duration#parse` should be able to parse it back 

### Actual behavior

an error is raised :  
```ruby
ActiveSupport::Duration::ISO8601Parser::ParsingError: Invalid ISO 8601 duration: "PT1e+09S" 
# at 
active_support/duration/iso8601_parser.rb:97:in `raise_parsing_error'
```

### System configuration
**Rails version**: Rails 6.0.3.2

**Ruby version**: ruby 2.6.5p114

### Informations
strangely, when specifying a precision, the bug does not appear
```ruby
big_number.seconds.iso8601(precision: 0) # "PT1000000000S"
```
Or when it comes to hours
```ruby
big_number.hours.iso8601 # "PT1000000000H"
```

**Comment by skipkayhil** [comment]
I believe this was fixed in #42348 (which was included in 7.0), can you confirm?

**Comment by stanig2106** [comment]
yes indeed, I am sorry for the inconvenience.
thank-you for your prompt response
